in_tail: Add descriptions for encoding parameters on in tail (#1870)
fluent/fluent-bit-docs: pipeline/inputs/tail.md

| `Static_Batch_Size` | Set the maximum number of bytes to process per iteration for the monitored static files (files that already exist upon Fluent Bit start). | `50M` |
| `File_Cache_Advise` | Set the `posix_fadvise` in `POSIX_FADV_DONTNEED` mode. This reduces the usage of the kernel file cache. This option is ignored if not running on Linux. | `On` |
| `Threaded` | Indicates whether to run this input in its own [thread](../../administration/multithreading.md#inputs). | `false` |

> | `Unicode.Encoding` | Set the encoding which the origin of character encoding. Currently, UTF-16LE, UTF-16BE, and auto is supported. | _none_ |

Author: Applied with rebasing the linked PR.

{% hint style="info" %}
If the database parameter `DB` isn't specified, by default the plugin reads each target file from the beginning. This might cause unwanted behavior. For example, when a line is bigger than `Buffer_Chunk_Size` and `Skip_Long_Lines` isn't turned on, the file will be read from the beginning of each `Refresh_Interval` until the file is rotated.
{% endhint %}

{% hint style="info" %}
Note that `Unicode.Encoding` depends on simdutf library which is written in C++11 or above.
So, the older platforms are not supported for this feature.
In addition, `Unicode.Encoding auto` is not covered for the all of the usages.
This is because sometimes this auto-detecting for character encodings makes a mistake to guess the correct encoding.
We recommend to use `UTF-16LE` or `UTF-16BE` if the target file encoding is pre-determined or known beforehand.
In details, this parameter requests to use 2-bytes aligned chunk and buffer sizes.
If they are not aligned for 2 bytes, Fluent Bit will use 2-bytes alignments automatically to avoid character breakages on consuming boundaries.

> {% endhint %}

Author: Applied with rebasing the linked PR.

- Final note: the `Path` patterns can't match the rotated files. Otherwise, the rotated file would be read again and lead to duplicate records.

{% endhint %}

## Character Encoding Conversion

This feature allows Fluent Bit to convert logs from various character encodings into the standard UTF-8 format.
This is crucial for processing logs from systems, especially Windows, that use legacy or non-UTF-8 encodings.
Proper conversion ensures that your log data is correctly parsed, indexed, and searchable.

### When to Use This Feature

> You should use this feature if your log files or messages are not in UTF-8 and you are seeing garbled or incorrectly rendered characters.

Reviewer: ```suggestion
You should use this feature if your log files or messages aren't in UTF-8 and you are seeing garbled or incorrectly rendered characters.
```

This is common in environments that use:

- Modern Windows applications that log in UTF-16.

> - Legacy Windows systems with applications that use traditional code pages (e.g., ShiftJIS, GBK, Win1252).

Reviewer: ```suggestion
- Legacy Windows systems with applications that use traditional code pages (for example, ShiftJIS, GBK, Win1252).
```

### Configuration Parameters

To enable encoding conversion, you will use one of the following two parameters within an input plugin configuration.

1. `Unicode.Encoding`

> Use this parameter for high-performance conversion of UTF-16 encoded logs to UTF-8. This method utilizes modern processor features (SIMD instructions) to accelerate the conversion process, making it highly efficient.

Reviewer: ```suggestion
   Use this parameter for high-performance conversion of UTF-16 encoded logs to UTF-8. This method utilizes modern processor features (SIMD instructions) to accelerate the conversion process, making it highly efficient. 
```

These have to be indented properly or the numbering doesn't work.

- Use Case: Ideal for logs coming from modern Windows environments that default to UTF-16.
- Supported Values:
  - UTF-16LE (Little-Endian)
  - UTF-16BE (Big-Endian)

1. `Generic.Encoding`

Use this parameter to convert from a wide variety of other character encodings, particularly legacy Windows code pages.

- Use Case: Essential for logs from older systems or applications configured for specific regions, common in East Asia and Eastern Europe.
- Supported Values: You can use any of the names or aliases listed below.

### East Asian Encodings

- `ShiftJIS` (Aliases: `SJIS`, `CP932`, `Windows-31J`)
- `GB18030`
- `GBK`: (Alias: `CP936`)
- `UHC` (Unified Hangul Code): (Aliases: `CP949` and `Windows-949`)
- `Big5`: (Alias: `CP950`)

> ### Windows (ANSI) Encodings

Reviewer: ```suggestion
### Windows (ANSI) encodings
```
